### PR TITLE
[ACS-5738] Make libraries in datatable to be possible to unselect

### PR DESCRIPTION
**Pull request by DominikIwanek**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Libraries on My Libraries page cannot be unselected.
Selection changes only on selecting another library.

**What is the new behaviour?**

User is able to unselect libraries by clicking on a row again.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/src/lib/datatable/components/datatable/datatable.component.ts [hunk 1]
```diff
@@ -320,6 +320,7 @@ export class DataTableComponent implements OnInit, AfterContentInit, OnChanges,
 
         if (changes.selectionMode && !changes.selectionMode.isFirstChange()) {
             this.resetSelection();
+            console.log('UNSELECT');
```
Comment (MichalKinas) [inline lib/core/src/lib/datatable/components/datatable/datatable.component.ts:323]: Probably you forgot to remove this console.log?

File: lib/core/src/lib/datatable/components/datatable/datatable.component.ts [hunk 2]
```diff
@@ -550,7 +557,6 @@ export class DataTableComponent implements OnInit, AfterContentInit, OnChanges,
             }
         }
     }
-
```
Comment (MichalKinas) [inline lib/core/src/lib/datatable/components/datatable/datatable.component.ts:553]: Is this intentional?

File: lib/core/src/lib/datatable/components/datatable/datatable.component.ts [hunk 3]
```diff
@@ -527,9 +528,15 @@ export class DataTableComponent implements OnInit, AfterContentInit, OnChanges,
     private handleRowSelection(row: DataRow, e: KeyboardEvent | MouseEvent) {
         if (this.data) {
             if (this.isSingleSelectionMode()) {
-                this.resetSelection();
-                this.selectRow(row, true);
-                this.emitRowSelectionEvent('row-select', row);
+                const isRowSelected = row.isSelected
+                if (isRowSelected) {
```
Comment (MichalKinas) [inline lib/core/src/lib/datatable/components/datatable/datatable.component.ts:532]: If you use `isRowSelected` only in the if, you can just use `row.isSelected` directly without creating new variable

File: lib/core/src/lib/datatable/components/datatable/datatable.component.ts [hunk 4]
```diff
@@ -527,9 +528,15 @@ export class DataTableComponent implements OnInit, AfterContentInit, OnChanges,
...
+                    this.resetSelection();
```
Comment (MichalKinas) [inline lib/core/src/lib/datatable/components/datatable/datatable.component.ts:533]: This line is used in both cases so you can move it above the whole if statement

**Comment by swapnil-verma-gl** [comment]
I think there might have been an issue with the rebase? There are 8 contributors, and 237 changed files.

**Pull request closed without merging**